Src/String/String.cpp

```diff
@@ -16,6 +16,10 @@
 
 using namespace PGE;
 
+#define PGE_STRING_WITH_SIZE(ret, chs, data, size) \
+String ret(nullptr); \
+auto [chs, data] = ret.reallocate(size)
```

Comment:
Kill this macro

Reply:
Having some utility to allocate Strings with a given size seems useful.

Reply:
What made the constructor that fulfilled this purpose not appropriate?

Reply:
It's unable to provide the result of the reallocation